runtime/shared_heap.c

```diff
@@ -406,7 +417,11 @@ value* caml_shared_try_alloc(struct caml_heap_state* local, mlsize_t wosize,
 
   CAML_EV_ALLOC(wosize);
 
-  if (whsize <= SIZECLASS_MAX) {
+  if (whsize <= SIZECLASS_MAX
+    && 
+      /* Cache line on a modern CPU is half of SIZECLASS_MAX. Let's delegate 
+        this entire case to large_alloc. */
```

Comment:
If I understand correctly, this comment is also wrong because SIZECLASS_MAX is 128 words while cache lines are 64 bytes.